### PR TITLE
a thorough, automated, reproducible install and documentation of kube…

### LINKED ISSUE #5694: nginx deployment example on edge in pending due to Taint: network-unavailable:NoSchedule
**What happened**:
I did install KubeEdge in cloud (AKS) and on edge. All seems fine, also the mosquitto did autodeploy to edge node and `kubectl get nodes` reports to ready nodes:
```
NAME                                STATUS   ROLES        AGE   VERSION
aks-nodepool1-39443061-vmss000000   Ready    agent        15h   v1.28.9
nuciiotedge                         Ready    agent,edge   70m   v1.28.6-kubeedge-v1.17.0
```

The edge node however has `Taints:             node.kubernetes.io/network-unavailable:NoSchedule` as this command shows `kubectl describe node nuciiotedge |  grep Taints`. Thus deploying nginx pod according to this [docu](https://release-1-17.docs.kubeedge.io/docs/setup/install-with-binary/#deploy-demo-on-edge-nodes) hangs forever in pending state.

If I remove the taint: `kubectl taint node nuciiotedge node.kubernetes.io/network-unavailable:NoSchedule-` nginx gets deployed to the edge node as expected.
The edge node however gets tainted again immediately.

**What you expected to happen**:
No tainting of the edge node.

**How to reproduce it (as minimally and precisely as possible)**:
Follow installation instructions using keadm for KubeEdge and the try to deploy nginx as described.

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
```
Client Version: v1.30.2
Kustomize Version: v5.0.4-0.20230601165947-6ce0bf390ce3
Server Version: v1.28.9
WARNING: version difference between client (1.30) and server (1.28) exceeds the supported minor version skew of +/-1
```

- KubeEdge version(e.g. `cloudcore --version` and `edgecore --version`):
both 1.17.0

- <details><summary>Cloud nodes Environment:</summary>

  - Hardware configuration (e.g. `lscpu`):
  - OS (e.g. `cat /etc/os-release`):
  - Kernel (e.g. `uname -a`):
  - Go version (e.g. `go version`):
  - Others:

  </details>
- <details><summary>Edge nodes Environment:</summary>

  - edgecore version (e.g. `edgecore --version`): 1.16
  - Hardware configuration (e.g. `lscpu`): x86_64
  - OS (e.g. `cat /etc/os-release`): Ubuntu 22.04.4 LTS
  - Kernel (e.g. `uname -a`): Linux nuciiotedge 5.15.0-113-generic #123-Ubuntu SMP Mon Jun 10 08:16:17 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux
  - Go version (e.g. `go version`): not found
  - Others:

  </details>

### PR DESCRIPTION
**Pull request by tjhgit**
**What type of PR is this?**
/kind documentation
/kind failing-test

**What this PR does / why we need it**:

This pull request is for everyone trying to install kubeedge. While the current documentation tries its best, it is not very specific. This pull request provides scripts and configuration files to make the install of kubeedge maximally reproducible. To this end it automatically deploys 2 azure vms running flatcar linux. With minimal user interaction one is installed as master deploying `kind` and `cloudcore` component and the other as edge with `edgecore` component.

The whole installation process takes maximum of 10 min time. It also includes `uninstall` scripts.

This pull request will just help users with the install of kubeedge and the kubeedge developers can concentrate on making this install work.

**Which issue(s) this PR fixes**:

Fixes #5694
Actually it fixes the above issue, but another issue is still existing `kindnet-xxx` are constantly crashing. So need help to fix this final problem. Debugging information is included in the pull request.

**Special notes for your reviewer**:

Please try it out yourself. It is well documented and very easy and **please help me fix the last open issue with this install.**.

**Does this PR introduce a user-facing change?**:

```release-note
NONE.
```

### CODE REVIEW

**Comment by tjhgit** [comment]
I added also a deployment on k3s, even tested with deploying an `nginx` pod onto edge. All seems to work here, but `kube-system   pod/svclb-traefik-0a82a694-ww6kz              0/7     Pending`.

**Review by fisherxu — COMMENTED** [review]
Is the script only useful for azure flactcar or for all linux?

File: docs/install_azure_flatcar/kind/master_install.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+set -xe
+
+# Assert that the current working directory is /root
+if [ "$(pwd)" != "/root" ]; then
+    echo "Error: The script must be run from the /root directory."
+    exit 1
+fi
+
+# create kubernetes cluster
+kind create cluster --config=/home/core/kind-config.yaml
```
Comment (Shelley-BaoYue) [inline docs/install_azure_flatcar/kind/master_install.sh:12]: Using kind to create cluster and then install KubeEdge? Maybe https://github.com/kubeedge/kubeedge/blob/master/hack/local-up-kubeedge.sh has completed the similar work.

**Comment by tjhgit** [comment]
> Is the script only useful for azure flactcar or for all linux?

These set of scripts do everything from vm provisioning (1 master vm and 1 edge vm) to installing kubeedge on flatcar linux. In case one would like to use another linux, the scripts need to be adapted.

File: docs/install_azure_flatcar/kind/master_install.sh (hunk 1 above)
Comment (tjhgit) [reply docs/install_azure_flatcar/kind/master_install.sh:12]: @Shelley-BaoYue . This script install kubeedge : cloudcore and edgecore on the SAME machine and uses this approach: https://kubeedge.io/docs/setup/install-with-binary. My scripts provision 2 separate VMs and then install kubeedge using helm (for cloudcore) and keadm for edgecore. It helps to have an automated installation for this kind of use case of two VMs or one VM and one real edge device. Also it uses minimal size flatcar linux, which makes the install very fast.
